ocs_ci/ocs/elasticsearch.py

```diff
@@ -0,0 +1,198 @@
+                    break
+            except IndexError:
+                log.info('ECK operator pod not ready yet')
+
+    def get_ip(self):
+        """
+        This function return the IP address of the Elasticsearch cluster.
+        this IP is to use inside the OCP cluster
+
+        Return
+            str : String that represent the Ip Address.
+
+        """
+        return self.es.get()["spec"]["clusterIP"]
+
+    def get_port(self):
+        """
+        This function return the port of the Elasticsearch cluster.
+
+        Return
+            str : String that represent the port.
+
+        """
+        return self.es.get()["spec"]["ports"][0]["port"]
+
+    def _deploy_es(self):
+        log.info('Deploy the PVC for the ElasticSearch cluster')
+        self.ocp.apply(self.pvc)
+
+        log.info('Deploy the ElasticSearch cluster')
+        self.ocp.apply(self.crd)
+
+        for es_pod in TimeoutSampler(
+            300, 20, get_pod_name_by_pattern, 'quickstart-es-default', self.namespace
+        ):
+            try:
+                if es_pod[0] is not None:
+                    self.espod = es_pod[0]
+                    log.info(f'The ElasticSearch pod {self.espod} Started')
+                    break
+            except IndexError:
+                log.info('elasticsearch pod not ready yet')
+
+        es_pod = OCP(kind='pod', namespace=self.namespace)
+        log.info('Waiting for ElasticSearch to Run')
+        assert es_pod.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            resource_name=self.espod,
+            sleep=30,
+            timeout=600
+        )
+        log.info('Elastic Search is ready !!!')
+
+    def get_health(self):
+        """
+        This method return the health status of the Elasticsearch.
+
+        Returns:
+            bool : True if the status is green (OK) otherwise - False
+
+        """
+        return self.elasticsearch.get()['items'][0]['status']['health'] == 'green'
+
+    def get_password(self):
+        """
+        This method return the password used to connect the Elasticsearch.
+
+        Returns:
+            str : The password as text
+
+        """
+        return base64.b64decode(self.password.get()['data']['elastic']).decode('utf-8')
+
+    def cleanup(self):
+        """
+        Cleanup the environment from all Elasticsearch components, and from the
+        port forwarding process.
+
+        """
+        log.info('Teardown the Elasticsearch environment')
+        log.info(f'Killing the local server process ({self.lspid})')
+        os.kill(self.lspid, signal.SIGKILL)
+        log.info('Deleting all resources')
+        subprocess.run(f'oc delete -f {self.crd}', shell=True)
+        subprocess.run(f'oc delete -f {self.eck_file}', shell=True, cwd=self.dir)
+        self.ns_obj.wait_for_delete(resource_name=self.namespace)
+
+    def local_server(self):
+        """
+        Starting sub-process that will do port-forwarding, to allow access from
```

Comment:
The only way to connect the ES server from outside the OCP (e.g. the laptop that run the test), is by using this port-fwding.

Reply:
Discussed about this:  For jenkins slave this works cleanly and that is where we need port forwarding,  For individual runs, the  local portfwding is needed from port 9200, this can be disabled for individual runs since we care about data from jenkins and dont want to modify local fwding rules from laptop. Follow on PR should document this.